development/Dockerfile

```diff
@@ -1,44 +1,71 @@
     && poetry config installer.parallel false
 
+# !!! CAUTION MODIFYING LINES ABOVE
 # -------------------------------------------------------------------------------------
-# Install Nautobot Plugin
+# Plugin-specifc system build/test dependencies.
+# 
+# Example: LDAP requires `libldap2-dev` to be apt-installed before the Python package.
 # -------------------------------------------------------------------------------------
-WORKDIR /tmp/install
+# --> Start safe to modify section
 
-# Copy in only pyproject.toml/poetry.lock to help with caching this layer if no updates to dependencies
-COPY poetry.lock pyproject.toml /tmp/install/
+# Uncomment the line below if you are apt-installing any package. Extra step if no packages requries.
+# RUN apt update
+# RUN apt install libldap2-dev
 
-# Add the requested Nautobot version to pyproject
-# to install the correct version based on the NAUTOBOT_VER argument
-# Otherwise Poetry will override the version in this container
-# with the one in the poetry.lock
-RUN poetry add nautobot=${NAUTOBOT_VER}
-
-# --no-root declares not to install the project package since we're wanting to take advantage of caching dependency installation
-# and the project is copied in and installed after this step
-RUN poetry install --no-interaction --no-ansi --no-root
+# --> Stop safe to modify section
+# -------------------------------------------------------------------------------------
+# Install Nautobot Plugin
+# -------------------------------------------------------------------------------------
+# !!! CAUTION MODIFYING LINES BELOW
 
-# Copy in the rest of the source code and install local Nautobot plugin
+# Copy in the source code
 WORKDIR /source
 COPY . /source
-RUN cp /tmp/install/* /source/
-RUN poetry install --no-interaction --no-ansi
+
+# Get container's installed Nautobot version as a forced constraint
+RUN pip show nautobot | grep "Version: " | sed -e 's/Version: /nautobot==/' > constraints.txt
+
+# Use Poetry to grab dev dependencies from the lock file
+# Can be improved in Poetry 1.2 which allows `poetry install --only dev`
+#
+# We can't use the entire freeze as it takes forever to resolve with rigidly fixed non-direct dependencies,
+#   especially those that are only direct to Nautobot but the container included versions slightly mismatch
+RUN poetry export -f requirements.txt --without-hashes --output poetry_freeze_base.txt
+RUN poetry export -f requirements.txt --dev --without-hashes --output poetry_freeze_all.txt
+RUN sort poetry_freeze_base.txt poetry_freeze_all.txt | uniq -u > poetry_freeze_dev.txt
+
+# Install all local project as editable, constrained on Nautobot version, to get any additional
+#   direct dependencies of the plugin
+RUN pip install -c constraints.txt -e .
+
+# Install any dev dependencies frozen from Poetry
+# Can be improved in Poetry 1.2 which allows `poetry install --only dev`
+RUN pip install -c constraints.txt -r poetry_freeze_dev.txt
```

Comment:
Poetry 1.2 is in Nautobot Dev containers from 1.4.2 onward. Once that version becomes the lowest common denominator we can switch this to
```Dockerfile
RUN pip install -c constraints.txt -e .
RUN poetry install --only dev
```
but for any plugins still seeking to install on 1.3 of Nautobot we must take the scenic tour.

Reply:
Still not comfortable yet depending on Poetry to solve these challenges yet.